### ISSUE TITLE
Should lock use `internalOptions.request_type` on hosted page if set?

### ISSUE THREAD

**Issue by selaux** [auth0/lock#1281]
I found some confusing code on the current `Lock` default template for the hosted login form and just wanted to check in with you, whether my assumption is correct.

My Assumption: Lock should use the `response_type` from `config.internalOptions` on the hosted login page if set. It should NOT evaluate `config.callbackOnLocationHash` if `config.internalOptions.response_type` is set.

This is how I read it from the code of the template (especially because of the linebreaks):
```js
responseType: (config.internalOptions || {}).response_type ||
          config.callbackOnLocationHash ? 'token' : 'code',
```

As I extracted this part to a function and wrote tests for this, I noticed the following:

```js
const getResponseType = (config) => (config.internalOptions || {}).response_type ||
          config.callbackOnLocationHash ? 'token' : 'code';

// evaluates to 'code', seems correct
getResponseType({})
// evaluates to 'token', seems correct
getResponseType({ callbackOnLocationHash: true })
 // evaluates to 'token', SEEMS INCORRECT
getResponseType({ internalOptions: { response_type: 'code' } })
```

Basically if `config.internalOptions.response_type` is set. It's contents will be ignored and `"token"` will be used. This is because [operator precedence](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence) will cause `(config.internalOptions || {}).response_type ||
          config.callbackOnLocationHash` to be evaluated first and only afterwards it will apply the ternary operator to the result.

This could be fixed easily by putting parentheses around the ternary expression. But first my initial question: Is this intended?

**Comment by selaux** [comment]
Thx for the clarification.
